packages-ts/starknet-gauntlet/README.md

```diff
@@ -1 +1,116 @@
 # Starknet Gauntlet
+
+## Overview
+
+In the following guide, we'll provide a walk through on how to setup your environment to run Gauntlet.
+
+## Prerequisites
+
+First, you'll need to install the following software on your machine (please refer to the `.tool-versions` file in the repo's root directory for more info on the specific versions you should install):
+
+- [Node.js](https://nodejs.org/en/download/package-manager) and [yarn](https://classic.yarnpkg.com/lang/en/docs/install)
+- [Scarb](https://docs.swmansion.com/scarb/download.html)
+
+## Getting Started
+
+Once the necessary software has been installed, navigate to the root directory of the repo. From there, we can use the following command to build Gauntlet and compile both the Cairo and Solidity Starknet smart contracts:
+
+```sh
+make build-ts
+```
+
+Once everything has been built, you should be able to run Gauntlet:
+
+```sh
+yarn gauntlet
+```
+
+This should output a list of commands that come with Gauntlet:
+
+```sh
+yarn run v1.22.21
+$ node ./packages-ts/starknet-gauntlet-cli/dist/index.js
+.env not found
+🧤  gauntlet 0.3.1
+.env not found
+ℹ️   Available gauntlet commands:
+
+access_controller:
+   access_controller:declare
+   access_controller:deploy
+   access_controller:upgrade
+   access_controller:transfer_ownership
+   access_controller:accept_ownership
+   access_controller:declare:multisig
+   access_controller:deploy:multisig
+
+ocr2:
+   ocr2:deploy
+...
+```
+
+## Selecting an RPC URL
+
+Before executing a Gauntlet command, you'll need a Starknet RPC v7 URL. You can find a list of common ones [here](https://www.starknetjs.com/docs/next/guides/connect_network/). Take note of one of these URLs and save it for later - we'll need it when we move onto configuring environment variables for Gauntlet.
```

Comment:
I think op-core would be using our private RPCs for this

Reply:
Sounds good! I'll add a note on this

Reply:
you should create a new file in packages-ts/starknet-gauntlet-cli/networks called .env.mainnet .

you can initialize it with a publicly available RPC, but you should add instructions that tell them to change the .env.mainnet or .env.testnet file to a private RPC if they'd like.

Also, while you're doing that you should also change https://github.com/smartcontractkit/chainlink-starknet/blob/augustus.BCI-3299.build-report/packages-ts/starknet-gauntlet-cli/networks/.env.testnet#L1 to use v7.

Reply:
actually, you may not need to add the instructions to change .env.mainnet file if you envision op-core just using the .env file for their setup

Reply:
Yup - added some clarifications in Slack

Marking as resolved